File: src/exemplos/3_Fluxo/2_Repeticao/4-do-while.c
/**      @file: 4-do-while.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Exemplo do laço de repetição DO-WHILE. */

#include <stdio.h>

int main (){
    int N = 10;
    int i = 0;

    printf("do-while(i < %d)\n\t",N);
    i = 0;
    do {
        printf("%d ", i++); /* Atualização da condição de parada. */
    } while(i < N);

    printf("\n");

    return 0; /* Não houve erro. */
}

File: src/exemplos/3_Fluxo/2_Repeticao/4-do-while.py
#  -*- coding: utf-8 -*-
#       @file: 4-do-while.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Em Python, não há o laço de repetição DO-WHILE. Mas o
# comportamento pode ser replicado da seguinte forma:
#
# while True:
#     # iteração
#
#     if criterio_de_parada:
#         break


N = 10

print 'do-while(i < %d)' % N
print '\t',
i = 0  # Inicialização do critério de parada.
while True:
    print i,
    i += 1  # Atualização da condição de parada.

    if i >= N:
        break

print
